src/domain/useCase/addSslPair.go

```diff
@@ -3,26 +3,60 @@ package useCase
 import (
 	"errors"
 	"log"
+	"slices"
 
 	"github.com/speedianet/os/src/domain/dto"
 	"github.com/speedianet/os/src/domain/repository"
+	"github.com/speedianet/os/src/domain/valueObject"
 )
 
 func AddSslPair(
 	sslCmdRepo repository.SslCmdRepo,
+	vhostQueryRepo repository.VirtualHostQueryRepo,
 	addSslPair dto.AddSslPair,
 ) error {
-	err := sslCmdRepo.Add(addSslPair)
+	existingVhosts, err := vhostQueryRepo.Get()
+	if err != nil {
+		log.Printf("FailedToGetVhosts: %s", err.Error())
+		return errors.New("FailedToGetVhostsInfraError")
+	}
+
+	if len(existingVhosts) == 0 {
+		log.Printf("VhostsNotFound")
+		return errors.New("VhostsNotFound")
+	}
+
+	existingVhostsStr := []string{}
+	for _, vhost := range existingVhosts {
+		existingVhostsStr = append(existingVhostsStr, vhost.Hostname.String())
+	}
+
+	shouldReturnVhostError := false
+	if len(addSslPair.VirtualHosts) == 1 {
+		shouldReturnVhostError = true
+	}
+
+	validVhosts := []valueObject.Fqdn{}
+	for _, vhost := range addSslPair.VirtualHosts {
+		if !slices.Contains(existingVhostsStr, vhost.String()) {
+			log.Printf("VhostDoesNotExists: %s", vhost.String())
+			if shouldReturnVhostError {
+				return errors.New("VhostDoesNotExists")
+			}
+
+			continue
+		}
+
+		validVhosts = append(validVhosts, vhost)
+	}
+
+	addSslPair.VirtualHosts = validVhosts
```

Comment:
Pés pelas mãos aqui. Custei entender sua intenção. Você cria um shouldReturnVhostError e fala que ele tem que ser true caso a addSslPair.VirtualHosts tem apenas 1 elemento. Daí você faz um for para remover vhosts que não existem e daí verifica se a shouldReturnVhostError tá true pra dar um erro geral. Lógica hiper complexa, você tá fazendo duas coisas ao mesmo tempo e fazendo confusa.

Primeiro você precisa filtrar os vhosts que não existem, alimentando a array validSslVirtualHosts com os itens válidos. DEPOIS você verifica se len(validSslVirtualHosts) > 1, pois você só pode proceder se ao menos um dos vhosts é válido. Daí você pode usar o addSslPair.VirtualHosts = validSslVirtualHosts logo após.

Reply:
O que o posicionamento das coisas não causam numa lógica, em...
Sua observação foi super válida e eu entendi exatamente onde eu causei a complexidade. Antes de validar, eu preciso saber quais os vhosts são válidos. Afinal, se nenhum for válido, então é óbvio que eu retornarei erro. A complexidade excessiva aí é que eu, para não alterar a lógica de criar um laço e verificar um vhost de cada vez e decidir se retorno ou não erro, acabou gerando a necessidade inútil da criação de um variável.

Tudo isso poderia ter sido evitado se eu fizesse o processo lógico: itera sobre os vhosts, vê quem são os válidos e os guarda. Verifica se existe pelo menos um vhost válido, pois se não existir, deve ser retornado erro. Só nisso eu impeço de ter um return no meio de um for e evito a construção de um variável inútil e com um nome super complexo.

Consegui entender meu erro. Ajustei o código!